Journal/Journal/EntryController.swift

```diff
@@ -0,0 +1,89 @@
+//  Copyright © 2019 Frulwinn. All rights reserved.
+
+import Foundation
+
+let baseURL = URL(string: "https://books-118d9.firebaseio.com/")!
+
+class EntryController {
+    var entries: [Entry] = []
+    
+    func put(entry: Entry, completion: @escaping(Error?) -> Void) {
+
+        let url = baseURL.appendingPathComponent(entry.identifier)
+        let newURL = url.appendingPathExtension("json")
+        
+        var request = URLRequest(url: newURL)
+        request.httpMethod = "PUT"
+        
+        do {
+            let encoder = JSONEncoder()
+            request.httpBody = try encoder.encode(entry)
+        } catch {
+            print(error)
+            completion(error)
+            return
+        }
+        
+        URLSession.shared.dataTask(with: request) { (_, _, error) in
+            if let error = error {
+                print(error)
+                completion(error)
+                return
+            }
+            self.entries.append(entry)
+            completion(nil)
+        }.resume()
+    }
+    
+    func createEntry(title: String, bodyText: String, completion: @escaping(Error?) -> Void) {
+        let entry = Entry(title: title, bodyText: bodyText)
+        put(entry: entry, completion: completion)
+        
+    }
+    
+    func updateEntry(entry: Entry, title: String, bodyText: String, completion: @escaping(Error?) -> Void) {
+        guard let index = entries.index(of: entry) else { return }
+        entries[index].title = title
+        entries[index].bodyText = bodyText
+        put(entry: entry, completion: completion)
```

Comment:
Entry is a struct. It's a value type. When you assign it to a new property it make a new copy. So they both becomes two different unrelated objects. Your update function gets an entry then find its index from the array and change values of entry in the array. After that the entry inside of the array and the entry passed as the function's argument has different value. You need to send updated one to the server
```suggestion
        let updatedEntry = entries[index]
        put(entry: updatedEntry, completion: completion)
```